experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,309 @@
+
+function vertical_perm(o::Origami)
+  return o.v
+end
+
+function degree(o::Origami)
+  return o.d
+end
+
+function Base.show(io::IO, o::Origami)
+  h = horizontal_perm(o)
+  v = vertical_perm(o)
+  d = degree(o)
+  print(io, "Origami ($(h),$(v), $(d))")
+end
+
+GapObj(O::Origami) = O.o
+
+@doc raw"""
+    stratum(o::Origami)
+
+This function calculates the stratum of an origami, i.e. the list of the (non-ascendingly
+ordered) non-zero degrees of the singularities. For a singularity of cone angle 2kπ the
+degree of the singularity is k-1.
+
+# Examples
+```jldoctest
+julia> o = origami(cperm([1,2],[3,4]), cperm([2,5,3,6],[4,7]))
+Origami ((1,2)(3,4),(2,5,3,6)(4,7), 7)
+
+julia> stratum(o)
+3-element Vector{Int64}:
+ 2
+ 1
+ 1
+```
+"""
+function stratum(o::Origami)
+  h = horizontal_perm(o)
+  v = vertical_perm(o)
+  commutator = comm(h, v)
+  cycs = cycles(commutator)
+  unsorted_stratum = [length(c) - 1 for c in cycs if length(c) > 1]
+  return sort!(unsorted_stratum; rev=true)
+end
+
+@doc raw"""
+    genus(o::Origami)
+
+Calculates the genus of an origami.
+
+# Examples
+```jldoctest
+julia> o = origami(cperm([1,2]), cperm([2,3]))
+Origami ((1,2),(2,3), 3)
+
+julia> genus(o)
+2
+```
+"""
+function genus(o::Origami)
+  return ZZ((sum(stratum(o)) + 2) / 2)
+end
+
+function veech_group(O::Origami)
+  # TODO use hashtables or not? Implement modular subgroup?
+  GAP.Globals.ComputeVeechGroupWithHashTables(GapObj(O))
+end
+
+function index_monodromy_group(o::Origami)
+  return GAP.Globals.IndexOfMonodromyGroup(GapObj(o))::Int
+end
+
+function sum_of_lyapunov_exponents(o::Origami)
+  return QQFieldElem(GAP.Globals.SumOfLyapunovExponents(GapObj(o)))
+end
+
+function translations(o::Origami)
+  h = horizontal_perm(o)
+  v = vertical_perm(o)
+
+  function act_on_tuples_by_conj(tup::Vector{PermGroupElem}, g::PermGroupElem)
+    return [tup[1]^g, tup[2]^g]
+  end
+
+  S = stabilizer(symmetric_group(degree(o)), [h, v], act_on_tuples_by_conj)[1]
+  return elements(S)
+end
+
+function is_hyperelliptic(o::Origami)
+  # check whether -1 is in the veech group
+  if !veech_group_is_even(o)
+    return false
+  end
+
+  x = horizontal_perm(o)
+  y = vertical_perm(o)
+  g = genus(o)
+
+  L = point_reflections(o)
+  L = filter(i -> order(i) == 2, L)
+
+  isempty(L) && return false
+
+  degree_list = collect(1:degree(o))
+  for sigma in L
+    # fixpoints
+    b =
+      length(setdiff(degree_list, moved_points(sigma))) +
+      length(setdiff(degree_list, moved_points(sigma * x))) +
+      length(setdiff(degree_list, moved_points(sigma * y)))
+
+    p1 = sigma * x^-1 * y^-1
+    p2 = y * x * (x * y)^-1
+    for i in degree_list
+      if i^p1 == i^p2
+        b += 1
+      end
+    end
+
+    if 2 * g + 2 - b == 0
+      return true
+    end
+  end
+
+  return false
+end
+
+function cylinder_structure(o::Origami)
+  gap_obj = GAP.Globals.CylinderStructure(GapObj(o))
+  cyl_lists = Vector{Vector{Int}}(gap_obj)
+  cyl_tuples = [(l[1], l[2]) for l in cyl_lists]
```

Comment:
```suggestion
  cyl_tuples = Vector{Tuple{Int, Int}}(gap_obj)
```
this could work as well (I haven't tested it)